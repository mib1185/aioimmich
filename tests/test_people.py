"""Tests for aioimmich."""

from __future__ import annotations

from syrupy.assertion import SnapshotAssertion


async def test_get_all_people(mock_immich_with_data, snapshot: SnapshotAssertion):
    """Test async_get_all_tags."""
    api = await mock_immich_with_data()
    people = await api.people.async_get_all_people(page_size=3, max_pages=1)

    assert len(people) == 3
    assert people == snapshot
