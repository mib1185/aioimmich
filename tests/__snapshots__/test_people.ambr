# serializer version: 1
# name: test_get_all_people
  list([
    ImmichPerson(person_id='6176838a-ac5a-4d1f-9a35-91c591d962d8', is_hidden=False, name='Me', thumbnail_path='upload/thumbs/e7ef5713-9dab-4bd4-b899-715b0ca4379e/61/76/6176838a-ac5a-4d1f-9a35-91c591d962d8.jpeg', birth_date=None, color=None, is_favorite=False, updated_at=datetime.datetime(2025, 5, 11, 11, 7, 41, 651000, tzinfo=datetime.timezone.utc)),
    ImmichPerson(person_id='3e66aa4a-a4a8-41a4-86fe-2ae5e490078f', is_hidden=False, name='I', thumbnail_path='upload/thumbs/e7ef5713-9dab-4bd4-b899-715b0ca4379e/3e/66/3e66aa4a-a4a8-41a4-86fe-2ae5e490078f.jpeg', birth_date=None, color=None, is_favorite=False, updated_at=datetime.datetime(2025, 5, 19, 22, 10, 21, 953000, tzinfo=datetime.timezone.utc)),
    ImmichPerson(person_id='a3c83297-684a-4576-82dc-b07432e8a18f', is_hidden=False, name='Myself', thumbnail_path='upload/thumbs/e7ef5713-9dab-4bd4-b899-715b0ca4379e/a3/c8/a3c83297-684a-4576-82dc-b07432e8a18f.jpeg', birth_date=None, color=None, is_favorite=False, updated_at=datetime.datetime(2025, 5, 12, 21, 7, 4, 44000, tzinfo=datetime.timezone.utc)),
  ])
# ---
